### PR TITLE
[bitnami/mariadb] Don't overwrite existing secrets

### LINKED ISSUE #7403: [bitnami/mariadb] Don't overwrite generated secret values on chart upgrade
**Which chart**:
**Chart:** mariadb
**Version:** latest

**Describe the bug**
The `mariadb-password` attribute in the [charts secret](https://github.com/bitnami/charts/blob/9134e1700eebf6d0c9caaeb78e047cc7438eb064/bitnami/mariadb/templates/secrets.yaml) is getting overwritten on every upgrade, if the `mariadb-password` was [generated](https://github.com/bitnami/charts/blob/9134e1700eebf6d0c9caaeb78e047cc7438eb064/bitnami/mariadb/templates/secrets.yaml#L26).

**To Reproduce**

1. Intall the mariadb chart
2. Have a look at the secret
3. Upgrade the chart
4. Check the secret again => Entries have changed

**Expected behavior**
Wrap the lines [L19](https://github.com/bitnami/charts/blob/9134e1700eebf6d0c9caaeb78e047cc7438eb064/bitnami/mariadb/templates/secrets.yaml#L19), [L26](https://github.com/bitnami/charts/blob/9134e1700eebf6d0c9caaeb78e047cc7438eb064/bitnami/mariadb/templates/secrets.yaml#L26) and [L34](https://github.com/bitnami/charts/blob/9134e1700eebf6d0c9caaeb78e047cc7438eb064/bitnami/mariadb/templates/secrets.yaml#L34) into these template functions [`"common.secrets.passwords.manage"`](https://github.com/bitnami/charts/blob/9134e1700eebf6d0c9caaeb78e047cc7438eb064/bitnami/common/templates/_secrets.tpl#L61-L112) or [`"common.secrets.exists"`](https://github.com/bitnami/charts/blob/9134e1700eebf6d0c9caaeb78e047cc7438eb064/bitnami/common/templates/_secrets.tpl#L114-L129).

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.6.3", GitCommit:"d506314abfb5d21419df8c7e7e68012379db2354", GitTreeState:"dirty", GoVersion:"go1.16.5"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.1", GitCommit:"632ed300f2c34f6d6d15ca4cef3d3c7073412212", GitTreeState:"clean", BuildDate:"2021-08-19T15:38:26Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2+k3s1", GitCommit:"5a67e8dc473f8945e8e181f6f0b0dbbc387f6fca", GitTreeState:"clean", BuildDate:"2021-06-21T20:52:44Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
This probably applies to other charts as well (e.g. mysql, etc.).

### PR DESCRIPTION
**Pull request by mamiu**
**Description of the change**

Don't overwrite already generated secret values on mariadb chart upgrade

**Benefits**

MariaDB doesn't break on chart upgrade.

**Possible drawbacks**

None.

**Applicable issues**

  - fixes #7403 
  - fixes https://github.com/bitnami/bitnami-docker-mariadb/issues/225#issuecomment-880124912

**Additional information**

**Checklist**

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/mariadb/templates/secrets.yaml [hunk 1]
```diff
@@ -13,25 +13,23 @@ metadata:
...
-  mariadb-password: {{ randAlphaNum 10 | b64enc | quote }}
+  {{- if (not .Values.auth.forcePassword) }}
+  {{- $passwordValuePath := "" }}
+  {{- if not (empty .Values.auth.username) }}
+  {{- $passwordValuePath := "auth.password" }}
+  {{- end }}
+  mariadb-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "mariadb-password" "providedValues" (list $passwordValuePath) "context" $) }}
   {{- else }}
   mariadb-password: {{ required "A MariaDB Database Password is required!" .Values.auth.password }}
   {{- end }}
```
Comment (migruiz4) [inline bitnami/mariadb/templates/secrets.yaml:29]: I think we should reconsider this logic. If `auth.username` is not provided, this key will never be used so there is no point in generating it or forcing it to be filled if `auth.forcePassword=true`.

```suggestion
  {{- if (not (empty .Values.auth.username)) }}
  {{- if (not .Values.auth.forcePassword) }}
  mariadb-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "mariadb-password" "providedValues" (list "auth.password") "context" $) }}
  {{- else }}
  mariadb-password: {{ required "A MariaDB Database Password is required!" .Values.auth.password | b64enc | quote }}
  {{- end }}
  {{- end }}
```
```

File: bitnami/mariadb/templates/secrets.yaml [hunk 2]
```diff
@@ -13,25 +13,23 @@ metadata:
...
 type: Opaque
 data:
-  {{- if not (empty .Values.auth.rootPassword) }}
-  mariadb-root-password: {{ .Values.auth.rootPassword | b64enc | quote }}
-  {{- else if (not .Values.auth.forcePassword) }}
-  mariadb-root-password: {{ randAlphaNum 10 | b64enc | quote }}
+  {{- if (not .Values.auth.forcePassword) }}
+  mariadb-root-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "mariadb-root-password" "providedValues" (list "auth.rootPassword") "context" $) }}
   {{- else }}
   mariadb-root-password: {{ required "A MariaDB Root Password is required!" .Values.auth.rootPassword }}
```
Comment (migruiz4) [inline bitnami/mariadb/templates/secrets.yaml:19]:
```suggestion
  mariadb-root-password: {{ required "A MariaDB Root Password is required!" .Values.auth.rootPassword | b64enc | quote }}
```

File: bitnami/mariadb/templates/secrets.yaml [hunk 3]
```diff
@@ -13,25 +13,23 @@ metadata:
...
   {{- end }}
   {{- if eq .Values.architecture "replication" }}
-  {{- if not (empty .Values.auth.replicationPassword) }}
-  mariadb-replication-password: {{ .Values.auth.replicationPassword | b64enc | quote }}
-  {{- else if (not .Values.auth.forcePassword) }}
-  mariadb-replication-password: {{ randAlphaNum 10 | b64enc | quote }}
+  {{- if (not .Values.auth.forcePassword) }}
+  mariadb-replication-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "mariadb-replication-password" "providedValues" (list "auth.replicationPassword") "context" $) }}
   {{- else }}
   mariadb-replication-password: {{ required "A MariaDB Replication Password is required!" .Values.auth.replicationPassword }}
```
Comment (migruiz4) [inline bitnami/mariadb/templates/secrets.yaml:34]:
```suggestion
  mariadb-replication-password: {{ required "A MariaDB Replication Password is required!" .Values.auth.replicationPassword | b64enc | quote }}
```

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @mamiu,

Thank you very much for your contribution! Please check my comments below.

The sentences with `required` were missing `| b64enc | quote`. This bug never had a real impact in the previous logic because of the condition `{{- if not (empty .Values.auth.rootPassword) }}`, but now this line of code will be executed even if `Values.auth.rootPassword` is not empty.

Also I requested a change regarding the `mariadb-password` key.

File: bitnami/mariadb/templates/secrets.yaml (hunk 2 above)
Comment (mamiu) [reply bitnami/mariadb/templates/secrets.yaml]: True!

File: bitnami/mariadb/templates/secrets.yaml (hunk 3 above)
Comment (mamiu) [reply bitnami/mariadb/templates/secrets.yaml]: True as well!

File: bitnami/mariadb/templates/secrets.yaml (hunk 1 above)
Comment (mamiu) [reply bitnami/mariadb/templates/secrets.yaml:29]: Great catch! 👍

Just for me to understand: The current implementation will cause the helm installation to fail with "A MariaDB Database Password is required!" if `auth.username` is empty and `auth.forcePassword` is true. This won't be the case anymore with your suggestion. Is that meant to be?

**Review by mamiu — COMMENTED** [review]
Add the missing base64 encodings

**Comment by mamiu** [comment]
@migruiz4 Thanks again for your help! Much cleaner now. 👍

**Pull request merged**